Intersect.Server/Entities/Player.cs

```diff
@@ -3669,21 +3670,30 @@ public void CraftItem(Guid id)
                     }
 
                     //Give them the craft
-                    var quantity = Math.Max(mCraftItem.Quantity, 1);
-                    var itm = ItemBase.Get(mCraftItem.ItemId);
-                    if (itm == null || !itm.IsStackable)
+                    var quantity = Math.Max(craftItem.Quantity, 1);
+                    if (finalItem == null)
+                    {
+                        PacketSender.SendChatMsg(this, Strings.Errors.UnknownErrorTryAgain, ChatMessageType.Error, CustomColors.Alerts.Error);
+                        Log.Error($"Unable to find item descriptor {finalItem.Id} for craft {craftItem.Id}.");
+                        return;
+                    }
```

Comment:
You can move this to the top of the method directly beneath `item = ItemBase...`